### PR TITLE
test.py: topology raft not experimental

### PR DESCRIPTION
**Pull request by alecco**
Preparing for the switch of raft from experimental feature to dedicated option, adjust checks for topology tests.

While there, fix previous check.

### CODE REVIEW

**Comment by alecco** [comment]
@nyh `test/alternator/test_table.py` and `test/cql-pytest/conftest.py` should also be updated (and fixed).

File: test/topology/conftest.py [hunk 1]
```diff
@@ -182,12 +182,11 @@ def check_pre_raft(cql):
         return False
     # In Scylla, we check Raft mode by inspecting the configuration via CQL.
     experimental_features = list(cql.execute("SELECT value FROM system.config WHERE name = 'experimental_features'"))[0].value
-    return not '"raft"' in experimental_features
-
+    return 'raft' in experimental_features
```
Comment (nyh) [inline test/topology/conftest.py]: The name of this function "pre_raft" suggests (but perhaps it wasn't documented clearly) that the system has **pre-raft** (i.e., **before** raft) bugs - namely bugs in concurrent schema modification **in a single node cluster**. These bugs do not exist in Cassandra (again, on a single-node cluster) or in post-raft Scylla, hence we returned False when raft is enabled or in Cassandra - and True in Scylla when raft is not enabled.

So the function was correct, albeit confusing.

File: test/topology/conftest.py [hunk 2]
```diff
@@ -182,12 +182,11 @@ def check_pre_raft(cql):
...
 
 @pytest.fixture(scope="function")
 def fails_without_raft(request, check_pre_raft):
-    if check_pre_raft:
+    if not check_pre_raft:
```
Comment (nyh) [inline test/topology/conftest.py:192]: I'm afraid I don't understand this patch. Above you reversed the meaning of check_pre_raft, and here you reverse the result, so how does the whole patch change anything?

Actually, it's even worse. Your patch will break running on Cassandra: On Cassandra, your check_pre_raft still returns False, so `not check_pre_raft` will be true and the test will become xfail. But it shouldn't - Cassandra does not have our pre-raft bugs on single-node clusters.

**Review by nyh — CHANGES_REQUESTED** [review]

**Comment by nyh** [comment]
> @nyh `test/alternator/test_table.py` and `test/cql-pytest/conftest.py` should also be updated (and fixed).

Note that one of the nice things about in-tree tests is that they can be fixed in the same patch series that changes the code.
Isn't it easier (to backport) and nicer to just add these test-fixing patches to the same PR where you'll change the option name?

**Pull request closed without merging**